Mathlib/Analysis/Calculus/Deriv/Shift.lean

```diff
@@ -61,3 +61,33 @@ lemma deriv_comp_const_sub : deriv (fun x ↦ f (a - x)) x = -deriv f (a - x) :=
 
 lemma deriv_comp_sub_const : deriv (fun x ↦ f (x - a)) x = deriv f (x - a) := by
   simp_rw [sub_eq_add_neg, deriv_comp_add_const]
+
+section BigO
+
+open Topology Asymptotics Filter
+
+lemma ContinuousAt.isBigO {𝕜 𝕜' : Type*} [NormedRing 𝕜] [NormedRing 𝕜'] [NormOneClass 𝕜']
```

Comment:
This lemma doesn't seem very natural somehow. I expected that the conclusion would be `f =O[𝓝 z] (fun _ ↦ 1)`, and then the hypotheses can be far more general (the domain can be any topological space, etc). Taking a limit of `f (w + z)` at `w = 0`, rather than `f w` at `w = z`, makes things much more complicated. 

The proof can also be massively shortened: with the statement as you have written it, there is a one-line proof
```
lemma ContinuousAt.isBigO {𝕜 𝕜' : Type*} [NormedRing 𝕜] [NormedRing 𝕜'] [NormOneClass 𝕜']
    {f : 𝕜 → 𝕜'} {z : 𝕜} (hf : ContinuousAt f z) :
    (fun w ↦ f (w + z)) =O[𝓝 0] (fun _ ↦ (1 : 𝕜')) := 
  (ContinuousAt.comp ((zero_add z).symm ▸ hf) (by fun_prop)).tendsto.isBigO_one 𝕜'
```

Reply:
Thanks for the better proof! I'm not that well-versed in this part of the library...

You need a `1` in the target with the right properties; this is why the `[NormedRing 𝕜'] [NormOneClass 𝕜']` assumptions are there. My use case is something like
```lean
(fun x ↦ LFunction χ (↑x + (1 + I * ↑y))) =O[𝓝 0] fun y ↦ 1
```
(where `x` and `y` are real), that's why I wanted the version with a shift.

I think I need these two lemmas each only once in the later development, so I can also inline them.

Reply:
I think inlining them is the way to go (and hopefully the shorter proofs that Jireh and I have suggested will make that a lot less painful).